delegation/delegated.gno

```diff
@@ -0,0 +1,252 @@
+package delegation
+
+import (
+	"std"
+
+	"gno.land/p/demo/grc/grc20"
+	u256 "gno.land/p/demo/uint256"
+)
+
+// IDelegatedToken defines the interface for a delegated token.
+type IDelegatedToken interface {
+	GetStaker() std.Address
+	GetDelegatedTo(owner std.Address) std.Address
+	Delegate(owner, to std.Address)
+	Deposit()
+	DepositAmount(amount *u256.Uint)
+	Withdraw()
+	WithdrawAmount(amount *u256.Uint)
+}
+
+// DelegatedToken represents a token that can be delegated to another address.
+type DelegatedToken struct {
+	// staker is the address of the entity staking the tokens.
+	staker      std.Address
+
+	// token is the unerlying GRC20 token being staked.
+	token       grc20.Token
+
+	// delegatedTo maps owner address to their delegate address.
+	delegatedTo map[std.Address]std.Address
+
+	// balances keeps track of the delegated token balances for each account.
+	balances    map[std.Address]*u256.Uint
+
+	// allowances tracks the spending allowances for delegated tokens.
+	allowances  map[std.Address]map[std.Address]*u256.Uint
+
+	// totoalSupply represents the total amount of delegated tokens in circulation.
+	totalSupply *u256.Uint
+
+	// name is the name of the delegated token.
+	name        string
+
+	// symbol is the symbol of the delegated token.
+	symbol      string
```

Comment:
Seems like this struct includes metadata from grc20 spec. In this case please include decimals too.

+ Do we need getter for this or not?? I'm not sure

Reply:
both added https://github.com/gnoswap-labs/governance/pull/9/commits/ea81900449d5679339843915a4ccd666908041ec